server/procedure_collection.cc

```diff
@@ -99,6 +100,45 @@ initRoutes(RouteManager & manager)
     ProcedureRunCollection::initRoutes(*runManager);
 
     manager.childRoutes["runs"] = runManager;
+
+  RestRequestRouter::OnProcessRequest getLatestRun =
+      [=] (RestConnection & connection, const RestRequest & req,
+           const RestRequestParsingContext & context)
+    {
+        // load current procedure
+        Utf8String redirect("/v1/procedures/");
+        auto collection = manager.getCollection(context);
+        auto key = manager.getKey(context);
+        auto procedure = static_cast<Procedure *>(
+            collection->getExistingEntry(key).get());
+        redirect += encodeUriComponent(restEncode(key));
+
+        // load all runs
+        auto runs = procedure->runs.get();
+        auto keys = runs->getKeys();
+        if (keys.empty()) {
+            throw HttpReturnException(404, "not found");
+        }
+
+        // Find the latest run
+        Date latestRunDate = Date::negativeInfinity();
```

Comment:
One thing to clarify, as implemented the latestrun is the one that started last not the one that ended last.  As a consequence, the latest run might still be running.  Is this ok with the rest of the code?